commons/src/main/java/org/infinispan/commons/util/DistinctFunction.java

```diff
@@ -0,0 +1,17 @@
+package org.infinispan.commons.util;
+
+import java.util.function.Function;
+
+/**
+ * This is a marker interface to be used with {@link Function} which signals to some implementors that
+ * the function returns distinct values.  This can be helpful because when a function is applied to data is ambiguous
+ * if the resulting data produced is distinct or not.  This allows some callers of this method to apply additional
+ * performance optimizations taking this into account.
+ * <p>
+ * If a <b>function</b> is implemented with this and it doesn't produce distinct values, the operation of the
+ * consumer of this function may be undefined.
+ * @author wburns
+ * @since 9.0
+ */
+public interface DistinctFunction<T, R> {
```

Comment:
Sorry I snuck this one in cause I will want it for dist sorting and it makes sense here too :)